src/fsharp/FSharp.Core/array2.fs

```diff
@@ -130,17 +131,27 @@ namespace Microsoft.FSharp.Collections
             init (length1 array) (length2 array) (fun i j -> array.[b1+i,b2+j])
 
         [<CompiledName("CopyTo")>]
-        let blit (source : 'T[,])  sourceIndex1 sourceIndex2 (target : 'T[,]) targetIndex1 targetIndex2 count1 count2 = 
+        let blit (source : 'T[,])  sourceIndex1 sourceIndex2 (target: 'T[,]) targetIndex1 targetIndex2 count1 count2 = 
```

Comment:
I gather that error message will now report bound per dimensions when there is a mismatch, should tests be added to assert this behaviour does what is intended wrt. new error messages?